### ISSUE TITLE
[Bug]: Node 22 `punycode` and `fs.Stats` is deprecated error message

### ISSUE THREAD

**Issue by cloudyparts** [Zhengqbbb/cz-git#178]
### 💭 Describe the feature

# [punycode](https://nodejs.org/docs/latest-v22.x/api/punycode.html)
> Deprecated since: v7.0.0

Source Code: [lib/punycode.js](https://github.com/nodejs/node/blob/v22.4.1/lib/punycode.js)

The version of the punycode module bundled in Node.js is being deprecated. In a future major version of Node.js this module will be removed. Users currently depending on the punycode module should switch to using the userland-provided [Punycode.js](https://github.com/bestiejs/punycode.js) module instead. For punycode-based URL encoding, see [url.domainToASCII](https://nodejs.org/docs/latest-v22.x/api/url.html#urldomaintoasciidomain) or, more generally, the [WHATWG URL API](https://nodejs.org/docs/latest-v22.x/api/url.html#the-whatwg-url-api).

### 💡 Proposed Solution

Please replace with [Punycode](https://github.com/mathiasbynens/punycode.js)

**Comment by cloudyparts** [comment]
I'm currently using this as a workaround for Node 22.2

```bash
NODE_OPTIONS="--no-deprecation" npx czg
```

**Comment by Zhengqbbb** [comment]
Could u provide node version and start up error message

**Comment by cloudyparts** [comment]
Thanks for reviewing, let me know if you want/need more information.

```bash
(node:91890) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:398:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:337:10)
    at loadBuiltinModule (node:internal/modules/helpers:96:7)
    at Module._load (node:internal/modules/cjs/loader:1037:17)
    at Module.require (node:internal/modules/cjs/loader:1271:19)
    at require (node:internal/modules/helpers:123:16)
    at /xxxxxxx/node_modules/czg/lib/index.js:104:268891
    at /xxxxxxx/node_modules/czg/lib/index.js:1:663
    at /xxxxxxx/node_modules/czg/lib/index.js:104:285695
czg@1.9.3
```

```bash
❯ node --version
v22.2.0
```

**Comment by Zhengqbbb** [comment]
Upstream `node-fetch` error, I'll see if there is a way to deal with it.
https://github.com/node-fetch/node-fetch/issues/1797
https://github.com/openai/openai-node/issues/527

**Comment by Zhengqbbb** [comment]
Thank you for pointing out the issue with this node@22 version. 
I have currently released a pre-release version `1.9.4-beta.1` for testing. 
I plan to conduct tests over the weekend. 
During this period, you can try it out and discover more issues. Thanks again.

```sh
npm install -g czg@next
```

![CleanShot 2024-07-15 at 20 55 35@2x](https://github.com/user-attachments/assets/f0476447-bdd5-4338-86dd-4b35fbafae4e)

**Comment by cloudyparts** [comment]
Thank you for the quick response!  This resovled my issue.
